### PR TITLE
Consume host resolver source package from dotnet/runtime

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses #22843 FYI source link will be broken until discussion at https://github.com/dotnet/runtime/pull/37793 is resolved.

### CODE REVIEW

**Comment by dougbu** [comment]
Should I be surprised this builds❔ I thought the package was missing a .targets file needed to include the file in compilation.

**Comment by JunTaoLuo** [comment]
The source is included without needing the .targets files. However, source link won't work without the .target file added.

**Comment by dougbu** [comment]
Got it. As long as the plan is for source link to work before RTM (preferably sooner), we're good.

**Comment by JunTaoLuo** [comment]
The work is tracked in https://github.com/dotnet/runtime/issues/37921. Is there anything blocking this PR @dougbu ?

File: src/Hosting/TestHost/src/Microsoft.AspNetCore.TestHost.csproj [hunk 1]
```diff
@@ -12,8 +12,7 @@
   <ItemGroup>
     <Reference Include="Microsoft.AspNetCore.Hosting" />
     <Reference Include="System.IO.Pipelines" />
-
-    <Compile Include="$(SharedSourceRoot)HostFactoryResolver\*.cs" />
+    <Reference Include="Microsoft.Extensions.HostFactoryResolver.Sources" />
```
Comment (Tratcher) [inline src/Hosting/TestHost/src/Microsoft.AspNetCore.TestHost.csproj:15]: Why does TestHost reference this? It doesn't do Startup discovery.

Comment (JunTaoLuo) [reply src/Hosting/TestHost/src/Microsoft.AspNetCore.TestHost.csproj:15]: https://github.com/dotnet/aspnetcore/blob/master/src/Hosting/TestHost/src/WebHostBuilderFactory.cs#L14

**Review by dougbu — APPROVED** [review]
> Is there anything blocking this PR @dougbu ?

No 😺

File: src/Hosting/TestHost/src/Microsoft.AspNetCore.TestHost.csproj (hunk 1 above)
Comment (Tratcher) [reply src/Hosting/TestHost/src/Microsoft.AspNetCore.TestHost.csproj:15]: Hmm, that literally isn't used anywhere.

**Pull request merged**